### PR TITLE
Feat: Poll accounts in NnsAccounts

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

We want to pollAccounts when visiting the NnsAccounts. I don't remember why I removed the call when the tokens page was enabled, but it's not correct. We want to poll accounts also there.

This helps only in edge cases when the initial loading of the accounts fails. Probably why we haven't seen any user report about it.

# Changes

* Enable `pollAccounts` in NnsAccounts when ENABLE_MY_TOKENS is enabled also.

# Tests

* Use the mocked API data instead of setting the store. This also tests that `pollAccounts` works as expected in NnsAccounts.

# Todos

- [ ] Add entry to changelog (if necessary).

Not necessary.

### CODE REVIEW

**Comment by lmuntaner** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #4411** <a href="https://app.graphite.dev/github/pr/dfinity/nns-dapp/4411?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dfinity/nns-dapp/4411?utm_source=stack-comment).

File: frontend/src/tests/mocks/icp-accounts.store.mock.ts [hunk 1]
```diff
@@ -65,8 +65,7 @@ export const mockSubAccountDetails: SubAccountDetails = {
 export const mockHardwareWalletAccountDetails: HardwareWalletAccountDetails = {
   name: "ledger test",
   principal: hardwareWalletPrincipal,
-  account_identifier:
-    "646f4d2d6fcb6fab5ba1547647526b666553467ecb5cb28c8d9ddf451c8f4c21",
+  account_identifier: mockHardwareWalletAccount.identifier,
```
Comment (lmuntaner) [inline frontend/src/tests/mocks/icp-accounts.store.mock.ts:68]: This is not necessarily needed because the identifier was the same. But it's important that they were both the same.

That's why I set it here based on the other mocks. We have the same pattern for the subaccount and main account details.

**Comment by lmuntaner** [comment]
@dskloetd please review

File: frontend/src/tests/lib/routes/Accounts.spec.ts
Comment (dskloetd) [inline frontend/src/tests/lib/routes/Accounts.spec.ts:1]: There seem to be a lot of changes here and it's not clear to me that they are all related to the polling change.
Did you do a bunch of unrelated test cleanups as well or were these all necessary?
If they are unrelated, it would be much easier to review in a separate PR/commit.

Comment (lmuntaner) [reply frontend/src/tests/lib/routes/Accounts.spec.ts:1]: The `pollAccounts` doesn't trigger if the store is full.

Therefore, when I removed setting up the store and all tests failed.

To fix it, I mocked the calls that would be used to fill up the store.

File: frontend/src/tests/lib/routes/Accounts.spec.ts [hunk 2]
```diff
@@ -130,15 +131,21 @@ vi.mock("$lib/services/worker-balances.services", () => ({
 
 describe("Accounts", () => {
   const balanceIcrcToken = 314000000n;
-  const newSubaccountName = "test";
+  const subaccountName = "test";
```
Comment (lmuntaner) [inline frontend/src/tests/lib/routes/Accounts.spec.ts:134]: Because now the subaccount might come from the backend, I renamed it and use it when the subaccount is present from the beginning.

File: frontend/src/tests/lib/routes/Accounts.spec.ts [hunk 3]
```diff
@@ -177,21 +187,30 @@ describe("Accounts", () => {
...
+          ? {
+              sub_accounts: [subaccountDetails],
+            }
+          : {}),
+        ...(hardwareWalletDetails
+          ? {
+              hardware_wallet_accounts: [hardwareWalletDetails],
+            }
+          : {}),
+      };
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/tests/lib/routes/Accounts.spec.ts:213]: Makes sense, done.

File: frontend/src/tests/lib/routes/Accounts.spec.ts [hunk 4]
```diff
@@ -614,32 +626,24 @@ describe("Accounts", () => {
...
 
       it("user can add a new account", async () => {
-        icpAccountsStore.setForTesting({
-          main: {
-            ...mockMainAccount,
-            balanceUlps: mainAccountBalance,
-          },
-          subAccounts: [],
-          hardwareWallets: [],
-        });
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/tests/lib/routes/Accounts.spec.ts:641]: Because I didn't want to assume that the accounts are in the store from the beginning.

This page should work without the store being full. That's what we are enabling with `pollAccounts` and what is being tested as well.

**Comment by lmuntaner** [comment]
@dskloetd ready for another reviw

**Review by dskloetd — APPROVED** [review]
